course5/homework6/string.cpp

```diff
@@ -0,0 +1,117 @@
+#include <memory>
+#include <cstring>
+#include <iostream>
+
+class String
+{
+public:
+    String() { }
+
+    String(const String& other)
+        : buffer_(std::make_unique<char[]>(sizeof(other.buffer_)))
+        , size_(other.size_)
+    {
+        std::uninitialized_copy_n(other.buffer_.get(), size_, buffer_.get());
+    }
+
+    String(const char* str)
+        : buffer_(std::make_unique<char[]>(sizeof(str)))
+        , size_(std::strlen(str))
+    {
+        std::uninitialized_copy_n(str, size_, buffer_.get());
+    }
+
+    String(String&& other)
+        : buffer_(std::move(other.buffer_))
+        , size_(other.size_)
+    {
+        other.size_ = 0;
+    }
+
+    String& operator=(const String& other) = default;
```

Comment:
Ha! I knew it! I doesn't actually compile. The caveat is: the compiler will try to generate the function only when you are using it. And you are not, which exposes a hole in your test coverage.